src/silx/gui/plot3d/_model/items.py

```diff
@@ -339,7 +339,7 @@ def setData(self, column, value, role):
         if column == 0 and role == qt.Qt.CheckStateRole:
             item = self.item()
             if item is not None:
-                item.setVisible(value == qt.Qt.Checked)
+                item.setVisible(qt.Qt.CheckState(value) == qt.Qt.Checked)
```

Comment:
`qt.Qt.CheckState(value)` is needed or the comparison always fails with PyQt6/PySide6.

Reply:
Funny.